src/Basque/Mews.Fiscalizations.Basque.Tests/Bizkaia/LROEResponseDeserializationTests.cs

```
@@ -0,0 +1,77 @@
+
+            Assert.True(response.Registros.All(registro => registro.SituacionRegistro.EstadoRegistro.Equals(EstadoRegistroEnum.Correcto)));
+        });
+    }
+
+    [Test]
+    public void PartiallyCorrectResponse_Deserialization_Succeeds()
+    {
+        Assert.DoesNotThrow(() =>
+        {
+            LROEPJ240FacturasEmitidasConSGAltaRespuesta response = XmlSerializationHelper.
+            Deserialize<LROEPJ240FacturasEmitidasConSGAltaRespuesta>(PartiallyCorrectResponseFilename);
+
+            Assert.IsNotNull(response);
+            Assert.True(response.Registros.Length == NumberOfRecords);
+
+            //first registro is correct
+            var firstRecord = response.Registros.First();
+            Assert.True(firstRecord.SituacionRegistro.EstadoRegistro.Equals(EstadoRegistroEnum.Correcto));
+
+            //second registro is incorrect
+            var secondRecord = response.Registros.Last();
+            Assert.True(secondRecord.SituacionRegistro.EstadoRegistro.Equals(EstadoRegistroEnum.Incorrecto));
+        });
+    }
+
+    [Test]
+    public void IncorrectResponse_Deserialization_Succeeds()
+    {
+        Assert.DoesNotThrow(() =>
+        {
+            LROEPJ240FacturasEmitidasConSGAltaRespuesta response = XmlSerializationHelper.
+            Deserialize<LROEPJ240FacturasEmitidasConSGAltaRespuesta>(IncorrectResponseFilename);
+
+            Assert.IsNotNull(response);
+            Assert.True(response.Registros.Length == NumberOfRecords);
+            Assert.True(response.Registros.All(registro => registro.SituacionRegistro.EstadoRegistro.Equals(EstadoRegistroEnum.Incorrecto)));
+        });
+    }
+
+    [Test]
+    public void WrongFileFormat_Deserialization_Fails()
+    {
+        Assert.Throws<InvalidOperationException>(() =>
+        {
+            XmlSerializationHelper.
+                Deserialize<LROEPJ240FacturasEmitidasConSGAltaRespuesta>(TicketBaiFilename);
+        });
+    }
```

I don't understand this test case, to me it looks like we're testing that System.Xml.Serialization throws an exception if it can't deserialize the given type and content

---

yep, it's just that deserializing a file that is not in the expected format should throw that exception